modules/tinymce/src/core/main/ts/api/commands/LinkCommands.ts

```diff
@@ -13,7 +13,7 @@ export const registerCommands = (editor: Editor): void => {
 
     if (Type.isObject(linkDetails) && Type.isString(linkDetails.href)) {
       // Spaces are never valid in URLs and it's a very common mistake for people to make so we fix it here.
-      linkDetails.href = linkDetails.href.replace(/ /g, '%20');
+      linkDetails.href = linkDetails.href.replace(/ /g, '%20').replace(/&amp;/g, '&');
```

Comment:
Is this really the right spot. Since the link dialog and the pasting a link acts differently I think it's somewhere else this should go. Maybe entity decode the clipboard contents before paste.
If you look in the paste/SmartPaste.ts it looks like it just takes the HTML as is and then uses that as the link instead of first entity decoding it if it comes form HTML. That means other HTML like things like < and > etc that needs to be encoded might also fail.

- [ ] Consider different approach

Reply:
One could argue that maybe the space fix here should be when input is made not in the low level command but it has been like for a long time.

Reply:
Yes, it's the presence of the space check that makes me feel that this is an appropriate place to put it. If it'd not been there I'd agree with you that it would not be the right place and instead move it a bit higher up in the stack where it comes from the paste.